worlds/sc2/Client.py

```diff
@@ -484,7 +484,7 @@ def calculate_items(ctx: SC2Context) -> typing.Dict[SC2Race, typing.List[int]]:
 
     # Upgrades from completed missions
     if ctx.generic_upgrade_missions > 0:
-        for race in SC2Race:
+        for race in (SC2Race.TERRAN, SC2Race.ZERG):
```

Comment:
Could this be better done by iterating over all races and doing a `continue` if `type_flaggroups[race].get("Upgrade")` returns None? Would mean this code doesn't have to be updated when Protoss upgrades are added, so it would be better controlled entirely from Items.py

Reply:
I'd check this too - if the race has an upgrade group. It's too easy to be missed when Protoss are added